public/static/plugins/perf/js/trend_chart.js

```diff
@@ -593,7 +593,23 @@ var drawSingleTrendChart = function(params) {
     })
 
     focusedText
-      .attr('y', function(d, i) { return opsLabelsY[i] })
+      .attr({'y': function(d, i) { return opsLabelsY[i] },
+             transform: function(){
+               // center the text for the last 20 items
+               var x = 0;
+               if(series){
+                 var percent = idx * 100.0 /  series.length;
+                 if(percent< 33){
+                   x = 0;
+                 } else if ( percent >= 33 && percent < 66) {
+                   x = -(cfg.focus.labelOffset.x + this.getBBox().width / 2);
+                 } else {
+                   x = -(cfg.focus.labelOffset.x + this.getBBox().width);
```

Comment:
This comment is the most important - what if we calculate offset for each item with some sort of formula? The idea will be the same, but it will more smooth, without hopping at 33 - 34 and 66-67 percents. The code will be simpler.
What do you think?